spyder/utils/programs.py

```diff
@@ -1062,3 +1062,24 @@ def find_git():
         return find_program('git')
     else:
         return find_program('git')
+
+
+def get_user_env_variables():
+    """Return local environment variables"""
+    cmdstr = ''
+    if sys.platform == 'darwin':
+        cmdstr = ('[[ -e /etc/profile ]] && source /etc/profile; '
+                  '[[ -e ~/.bash_profile ]] && source ~/.bash_profile; '
+                  'printenv')
+    elif sys.platform.startswith('linux'):
+        pass
+    elif os.name == 'nt':
+        pass

```

Comment:
I think it's necessary to make this work too and @dalthviz doesn't have time to help you with that. See for instance:

https://stackoverflow.com/questions/5327495/list-all-environment-variables-from-the-command-line

Reply:
Okay, I'll take a look.